File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()

        right=len(nums)-1
        left=0
        max=0

        while right-left>0:
            temp=nums[right]+nums[left]
            if temp>max:
                max=temp

            left+=1
            right-=1

        return max
        

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var minPairSum = function(nums) {
    nums=nums.sort((a,b)=>a-b)
    let left=0
    let right=nums.length-1
    total=0

    while(right-left>0){
        const temp=nums[left]+nums[right]
        if(temp>total){
            total=temp
        }
        left++
        right--
    }
    return total
};